### PR TITLE
Hayden - Edges (C10) - JS-Adagrams

### PR DESCRIPTION
**Pull request by halcyon-and-on**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | all of the loops and conditionals in JS are essentially the same; the Math module was familiar but sadly some of the built-in ruby methods were missing from JS
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | yes, i mostly use MDN now instead of ruby-docs or whatever
What was a challenge you were able to overcome on this assignment? | not having the enumerable module!! i had to write my own shuffling / sampling algorithm from stackoverflow posts instead :(
What is your favorite thing about learning a new programming language? | it is so much fun with javascript in particular to finally be able to manipulate webpages without reloading them
What is your least favorite thing about learning a new programming language? | "Unexpected Token: Fatal" lol
Do you have any recommendations on how we could improve this project for the next cohort? | nope!

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | only one commit!!
**Code Requirements** |
`drawLetters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | x
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highestScoreFrom` pass | x
**Overall** |

Great work on this Hayden! The code looks great -- pretty much no syntax things I can nitpick at ;-) You did a wonderful job getting through this project; well done!

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,118 @@
...
+      currentIndex -= 1;
+
+      // And swap it with the current element.
+      temporaryValue = array[currentIndex];
+      array[currentIndex] = array[randomIndex];
+      array[randomIndex] = temporaryValue;
+    }
+
+    return array;
+  },
```
Comment (tildeee) [inline src/adagrams.js:19]: nice work on this!

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,118 @@
...
+
   drawLetters() {
-    // Implement this method for wave 1
+    const array = 'AAAAAAAAABBCCDDDDEEEEEEEEEEEEFFGGGHHIIIIIIIIIJKLLLLMMNNNNNNOOOOOOOOPPQRRRRRRSSSSTTTTTTUUUUVVWWXYYZ'.split('');
+    this.shuffle(array);
+    return array.slice(0, 10);
```
Comment (tildeee) [inline src/adagrams.js:24]: nicely done, but while you implemented this, maybe you can refactor it to just randomly pick ten elements (instead of shuffling the entire array and then getting the first ten)!
